packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -1158,12 +1159,15 @@ describe('<DataGrid /> - Filter', () => {
       expect(screen.queryByRole('tooltip')).to.equal(null);
 
       fireEvent.mouseOver(filterButton);
-      clock.tick(1000); // tooltip display delay
-
-      const tooltip = screen.getByRole('tooltip');
 
-      expect(tooltip).toBeVisible();
-      expect(tooltip.textContent).to.contain('"John" text string');
+      await waitFor(
+        () => {
+          const tooltip = screen.getByRole('tooltip');
+          expect(tooltip).toBeVisible();
```

Comment:
Is this assertion needed? 🤔 
If I'm not mistaken, `getByRole` would fail for both non-existant as well as visually hidden elements.

Reply:
I avoided changing things that I don't have to change in the context of removing the fake timers